app/models/parser/tiktok_item.rb

```diff
@@ -22,12 +23,26 @@ def parse_data_for_parser(doc, _original_url, _jsonld_array)
 
       handle_exceptions(StandardError) do
         @parsed_data[:raw][:api] = @parsed_data[:raw][:oembed] = get_tiktok_api_data(url)
-        match = url.match(TIKTOK_ITEM_URL)
+        if url.match(TIKTOK_ITEM_URL)
+          match = url.match(TIKTOK_ITEM_URL)
+          username = match['username']
+          external_id = match['id']
+          title = parsed_data['raw']['api']['title']
+        elsif url.match(TIKTOK_TAG_URL)  
+          username = ''
+          external_id = ''
+          title = "Tag: " + url.match(TIKTOK_TAG_URL)['tag']
```

Comment:
could you add a test here? seems like good behavior to capture. think we already catch the other cases through our existing tests

Reply:
@hartsick I tried adding two tests one to test if the 'Tag: ' is added, and another to test if `external_id` and `username` are empty strings for unmatched URL pattern.

The one for the tag seems to work, but the other one doesn't.

At first I tried writing something like this
```
test "sets 'external_id' and 'username' as empty string for unmatched URL pattern" do
    WebMock.stub_request(:get, 'https://www.tiktok.com/search?q=abcdef').to_return(status: 200, body: response_fixture_from_file('tiktok-item-page.html'))
    
    data = Parser::TiktokItem.new('https://www.tiktok.com/search?q=abcdef').parse_data(doc)

    assert_equal '', data['title']
    assert_equal '', data['username']
end
```

I got an error that told me to use the `oembed`. I changed it, and it passed.
But it failed on Travis, so I ran it again locally, and it's currently failing as well.
I don't know why it passed the first couple tries.

What is the correct way to write this? Should we be testing for this?